### PR TITLE
Upgrade timescaledb to latest released version (v1.7.0) and PG12 [BREAKING]

### PR DESCRIPTION
**Pull request by slowr**
### Description of PR

What component(s) does this PR affect?

- [x] Back-End (Database, Detection/Configuration/etc. Microservices)
- [ ] Front-End (Flask, API, UI, etc)
- [ ] Monitor (RIPE RIS, BGPStream RV/RIS/CAIDA, etc.)
- [ ] Docs (incl. wiki)
- [ ] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Detection/Configuration/etc. Microservices)
- [ ] Front-End (Flask, API, UI, etc)
- [ ] Monitor (RIPE RIS, BGPStream RV/RIS/CAIDA, etc.)
- [ ] Docs (incl. wiki)
- [ ] Build System

### Related Issue

Resolves #

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [x] This change requires a change in the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by vkotronis** [comment]
TODO: if production/dev upgrade fails, form proper upgrade process.

**Comment by slowr** [comment]
Dumping the data and restoring should be sufficient in order to do the upgrade. I will try to do the upgrade and restore from our backup.tar file.

**Comment by vkotronis** [comment]
@slowr does the migration check fail https://travis-ci.org/github/FORTH-ICS-INSPIRE/artemis/jobs/679542540 maybe because the message "database system is ready to accept connections" has changed in the newer version? Could we do more checks to make sure that the testing is robust and nothing breaks across migrations?

**Comment by slowr** [comment]
> @slowr does the migration check fail https://travis-ci.org/github/FORTH-ICS-INSPIRE/artemis/jobs/679542540 maybe because the message "database system is ready to accept connections" has changed in the newer version? Could we do more checks to make sure that the testing is robust and nothing breaks across migrations?

no the migration check fails because it cannot migrate from PG10 to PG12. The check is working correctly.

File: docker-compose.get-latest.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  echo "[!] Setup is already stopped.. this means we are going to use the daily backup and not the latest"
+  read -p "[?] Continue (y/[n])? " -n 1 -r
+  echo # (optional) move to a new line
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    upgrade
+  else
+    exit -1
+  fi
+else
+  docker-compose exec postgres sh -c 'pg_dump -d $POSTGRES_DB -U $POSTGRES_USER -F t -f /tmp/db.tar > /tmp/db.log 2>&1' && \
```
Comment (vkotronis) [inline docker-compose.get-latest.sh:42]: this should be backup instead of tmp, correct?

File: docker-compose.get-latest.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+function upgrade() {
+  echo "[+] Deleting all containers" && \
+  docker-compose down && \
```
Comment (vkotronis) [inline docker-compose.get-latest.sh:5]: this uses the default docker-compose.yaml, but what about the rest of the extra services yaml, e.g., the routinator one? Same for pulling, etc. Maybe the user should specify which yamls he is using somewhere in the script?

File: docker-compose.get-latest.sh (hunk 1 above)
Comment (slowr) [reply docker-compose.get-latest.sh:42]: for this version it's tmp as the volumes already existing are pointing to tmp

File: docker-compose.get-latest.sh (hunk 2 above)
Comment (slowr) [reply docker-compose.get-latest.sh:5]: I think we should make the users to have everything stopped already before upgrading.. so we just start the database->backup->stop->upgrade->restore

File: docker-compose.get-latest.sh (hunk 1 above)
Comment (vkotronis) [reply docker-compose.get-latest.sh:42]: Correct, so it is correct for this version (we need to change after users upgrade).

File: docker-compose.get-latest.sh (hunk 2 above)
Comment (vkotronis) [reply docker-compose.get-latest.sh:5]: Is there a command that stops all running instances, whatever the used yaml? Sth like force compose down whatever the "-f ..." used.

Comment (slowr) [reply docker-compose.get-latest.sh:5]: nope

Comment (vkotronis) [reply docker-compose.get-latest.sh:5]: Ok then just add an instruction to the users to shut down their running docker instances before proceeding (and leave also the default `docker-compose down` as you have it right now).

**Comment by slowr** [comment]
This will be release on a new minor version (v1.5.0) and not in a patch version. Putting this pull request to the backlog until we schedule the minor release.

**Comment by slowr** [comment]
I will reopen the PR when we want to upgrade the database version

**Pull request closed without merging**